packages/web3-eth-contract/test/integration/contract_events.test.ts

```diff
@@ -124,6 +124,45 @@ describe('contract', () => {
 				);
 			},
 		);
+
+		itIf(isWs)(
+			'should fetch past events when "fromBlock" is passed to contract.events.<eventName>',
+			async () => {
+				const eventValues = [11, 12, 13, 14];
+				// eslint-disable-next-line jest/no-standalone-expect
+				return expect(
+					processAsync(async resolve => {
+						// trigger multiple events
+						await Promise.all(
+							eventValues.map(v => {
+								// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+								return contractDeployed.methods
+									.firesMultiValueEvent('Event Value', v, false)
+									.send(sendOptions);
+							}),
+						);
```

Comment:
There is a need to wait for every transaction, before firing the next one, to prevent a possible nonce duplication.
```suggestion
						for (const eventValue of eventValues) {
							// Wait for every transaction, before firing the next one, to prevent a possible nonce duplication.
							// eslint-disable-next-line no-await-in-loop
							await contractDeployed.methods
								.firesMultiValueEvent('Event Value', eventValue, false)
								.send(sendOptions);
						}

```

Reply:
Yeap that did it, thank you!